<launch>
  <arg name="doDebug" default="false" />
  <arg if="$(arg doDebug)" name="launchPrefix" value="xterm -e gdb -args" />
  <arg unless="$(arg doDebug)" name="launchPrefix" value="" />
  
  <arg name="camera_name" default="primesense_camera" doc="Name of camera, to get properties" />
  <arg name="fps" default="10" doc="Max frames per second to report object detection results" />
  <arg name="image_topic_name" default="/camera/image" doc="camera topic name" />
  <arg name="show_debug_windows" default="true" doc="true => show debugging X-window panes" />
  <arg name="show_step_times" default="false" doc="true => show computation times for interesting parts of object recognition" />

  <arg name="low_hue_range" default="0" doc="Low range of hue" />
  <arg name="high_hue_range" default="44" doc="High range of hue" />
  <arg name="low_saturation_range" default="181" doc="Low saturation range" />
  <arg name="high_saturation_range" default="255" doc="High saturantion range" />
  <arg name="low_value_range" default="169" doc="Low value range" />
  <arg name="high_value_range" default="255" doc="High value range" />
  <arg name="low_contour_area" default="500" doc="Minimum contour area in square pixels" />
  <arg name="high_contour_area" default="180000" doc="Maximum contour area in square pixels" />

  <node pkg="cone_detector" name="cone_detector" type="cone_detector_node" respawn="false" output="screen" launch-prefix="$(arg launchPrefix)" >
    <param name="camera_name" value="$(arg camera_name)" type="str" />
    <param name="fps" value="$(arg fps)" type="int" />
    <param name="image_topic_name" value="$(arg image_topic_name)" type="str" />
    <param name="show_debug_windows" value="$(arg show_debug_windows)" type="bool" />
    <param name="show_step_times" value="$(arg show_step_times)" type="bool" />
    <param name="low_hue_range" value="$(arg low_hue_range)" type="int" />
    <param name="high_hue_range" value="$(arg high_hue_range)" type="int" />
    <param name="low_saturation_range" value="$(arg low_saturation_range)" type="int" />
    <param name="high_saturation_range" value="$(arg high_saturation_range)" type="int" />
    <param name="low_value_range" value="$(arg low_value_range)" type="int" />
    <param name="high_value_range" value="$(arg high_value_range)" type="int" />
    <param name="low_contour_area" value="$(arg low_contour_area)" type="int" />
    <param name="high_contour_area" value="$(arg high_contour_area)" type="int" />
  </node>
</launch>
